src/librustc_mir/borrow_check/diagnostics/region_errors.rs

```diff
@@ -53,13 +54,69 @@ impl ConstraintDescription for ConstraintCategory {
     }
 }
 
-#[derive(Copy, Clone, PartialEq, Eq)]
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
 enum Trace {
     StartRegion,
     FromOutlivesConstraint(OutlivesConstraint),
     NotVisited,
 }
 
+/// A collection of errors encountered during region inference. This is needed to efficiently
+/// report errors after borrow checking.
+crate type RegionErrors<'tcx> = smallvec::SmallVec<[RegionErrorKind<'tcx>; 4]>;
```

Comment:
Why is this a `SmallVec`? It's usually empty.

Reply:
Umm... that's exactly why it's a `SmallVec`? In the case were there is <5 errors in a single MIR Def, we don't allocate, and this is almost always (either because there is no error or because the number of errors is small).

Am I misunderstanding your question?

Reply:
Using less stack space for 0 errors seems more important to me than not allocating for 1-4 errors.

Reply:
I added a comment

Reply:
Ok, I changed it to a `Vec`.